Source/MapComponents/MapComponent_RenderManager.cs

```diff
@@ -476,16 +477,48 @@ private void SaveUnityEncoding(byte[] encodedImage)
             var filePath = CreateCurrentFilePath();
 
             File.WriteAllBytes(filePath, encodedImage);
+
             // Create tmp copy to file if needed
             if (!manuallyTriggered && PRModSettings.fileNamePattern == FileNamePattern.BothTmpCopy)
             {
                 File.Copy(filePath, CreateFilePath(FileNamePattern.Numbered, true));
             }
-
+            AdjustRenderSize(filePath);
             DoEncodingPost();
+
+        }
+
+        private void AdjustRenderSize(string filePath)
+        {
+            if (File.Exists(filePath))
+            {  
+                FileInfo RenderInfo = new FileInfo(filePath);
+                long RenderLenght = RenderInfo.Length / 1048576;
+                Log.Warning(RenderLenght.ToString());
+                
+                if (PRModSettings.qualityAdjustment == JPGQualityAdjustmentSetting.Automatic)
+                {
+                    if (RenderLenght > PRModSettings.renderSize)
+                    {
+                        if (PRModSettings.JPGQuality > 0)
+                        {
+                            PRModSettings.JPGQuality -= 1;
+                            Messages.Message("JPG quality decreased to " + PRModSettings.JPGQuality.ToString() + "% · Render size: " + RenderLenght.ToString() + " Target: " + PRModSettings.renderSize.ToString(), MessageTypeDefOf.CautionInput, false);
```

Comment:
I think setting it to automatic is for more advanced users, so they'll understand that it's because they put it to automatic. I think the feedback will be appreciated by most people.